crypto/bio/b_addr.c

```diff
@@ -830,8 +817,7 @@ int BIO_lookup(const char *host, const char *service,
 
                 if (se == NULL) {
 #ifndef OPENSSL_SYS_WINDOWS
-                    BIOerr(BIO_F_BIO_LOOKUP, ERR_R_SYS_LIB);
-                    ERR_add_error_data(1, hstrerror(h_errno));
+                    SYSerr(SYS_F_GETSERVBYNAME, errno);
```

Comment:
Because here we pass errno, and not h_errno. As discussed in #2091 getservbyname does *not* set h_errno. Linux man page says nothing about if it sets errno, while for example Solaris says so.